packages/caliper-ethereum/test/ethereum-connector.js

```diff
@@ -0,0 +1,75 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+const path = require('path');
+const sinon = require('sinon');
+const expect = require('chai').expect;
+const ConfigUtil = require('@hyperledger/caliper-core').ConfigUtil;
+const EthereumConnector = require('../lib/ethereum-connector');
+describe('EthereumConnector', function () {
+    let tempConfigFilePath;
+
+    beforeEach(() => {
+        tempConfigFilePath = path.resolve(
+            __dirname,
+            './sample-configs/networkconfig.json'
+        );
+        ConfigUtil.set(ConfigUtil.keys.NetworkConfig, tempConfigFilePath);
+    });
+
+    describe('EthereumConnector.installSmartContract', () => {
+        it('should deploy all contracts successfully when no privacy settings are used', async () => {
+            const workerIndex = 0;
+            const bcType = 'ethereum';
+            const ethereumConnector = new EthereumConnector(workerIndex, bcType);
+            const deployContractStub = sinon.stub(ethereumConnector, 'deployContract').resolves({
+                options: { address: '0x123' },
+            });
+            await ethereumConnector.installSmartContract();
+            sinon.assert.called(deployContractStub);
+            deployContractStub.restore();
+        });
+    });
+
+    describe('EthereumConnector.init', () => {});
+
+    describe('EthereumConnector.checkConfig()', function () {
```

Comment:
checkConfig is really a private method (should have been marked with an _ in the code base) and this is a good example of why 
1. we don't test private methods directly
2. we don't describe the test with it's method name
If we want to refactor this to show it is a private method we need to go in and change the test, both at implementation and decription points. It makes refactoring much easier with reduced work on the tests and the tests continue to prove that the refactor hasn't broken anything.

These tests should be incorporated into the testing of the construction of the EthereumConnector object via the factory which I think I mentioned last review

Reply:
Changes Made!!